@shared/api/internal.ts

```diff
@@ -310,6 +315,92 @@ export const confirmPassword = async (
   return response;
 };
 
+type BalancesResponse = {
+  contractId?: string;
+  issuer?: {
+    key: string;
+  };
+  decimals: number;
+  name: string;
+  symbol: string;
+  valueXdr: string;
+}[];
+
+export const getAccountIndexerBalances = async (
+  pubKey: string,
+  network: NETWORKS,
+): Promise<AccountBalancesInterface> => {
+  try {
+    const indexerUrl = INDEXER_URLS[network];
+    if (!indexerUrl) {
+      throw new Error("Indexer URL not found");
+    }
+    const contractIds = await getTokenIds(network);
+    const url = new URL(`${indexerUrl}/account-balances/${pubKey}`);
+    for (const id of contractIds) {
+      url.searchParams.append("contract_ids", id);
+    }
+    const response = await fetch(url.href);
+    const json = (await response.json()) as BalancesResponse;
+    const formattedBalances = json.map((balance) => {
+      const totalScVal = xdr.ScVal.fromXDR(
+        Buffer.from(balance.valueXdr, "base64"),
+      );
+      return {
+        ...balance,
+        total: scValToNative(totalScVal),
+      };
+    });
+
+    const balances = formattedBalances.reduce((prev, curr) => {
+      if (curr.symbol === "XLM") {
+        prev["native"] = {
+          token: { type: "native", code: "XLM" },
+          total: new BigNumber(curr.total),
+          available: new BigNumber(curr.total), // TODO: how to get available for xlm?
```

Comment:
I believe this should be `currentBalance - ((2 + accountBalances.subentryCount) * BASE_RESERVE)`?